### ISSUE TITLE
Add configuration for optional IPFS gateway url

### ISSUE THREAD

**Issue by MCTaylor17** [fission-suite/heroku-ipfs-ghost#12]
See: https://github.com/fission-suite/ghost-storage-adapter-ipfs/pull/6

**Comment by bmann** [comment]
@MCTaylor17 so, the default gateway URL should be https://ipfs.runfission.com. It's also part of the reason for the images downloading automatically when visiting the URL.

The https://runfission.com is actually our Web API, not our gateway. It CAN serve up individual hashes for files, but doesn't do anything for directories.

**Comment by bmann** [comment]
@MCTaylor17 I merged your PR, but please do the following:

- [ ] Make `ipfs.runfission.com` the default
- [ ] Ask for the env variable without /ipfs -- should be just top level path -- e.g. gateway.ipfs.io or ipfs.runfission.com

**Comment by MCTaylor17** [comment]
@bmann I checked a [list of gateways](https://ipfs.fooock.com/) (the ["normal" list](https://ipfs.github.io/public-gateway-checker/) is currently 404) and only 1 out 29 didn't use `/ipfs`.   That makes it *almost* standard spec (if it isn't already).

With that said, I can check/append `/ipfs` either here or in the adapter itself.  I'm already doing some [minor URL normalization over there](https://github.com/fission-suite/ghost-storage-adapter-ipfs/blob/5db0b8d6ab0c74844210c4e48fe8e7dbe5f85570/utils.js#L54) so I'm inclined to do it there.

Do you have any thoughts on that?

**Comment by bmann** [comment]
Yes, they all use /ipfs -- but that's the server path, not the host, and yes, /ipfs should be appended in the storage adapter.

**Comment by MCTaylor17** [comment]
Just to make sure I'm making the right changes, the storage adapter [currently uses ipfs.io as the default](https://github.com/fission-suite/ghost-storage-adapter-ipfs/blob/5db0b8d6ab0c74844210c4e48fe8e7dbe5f85570/index.js#L6) based on a requirement pulled out of the [original discourse description](https://talk.fission.codes/t/fission-ipfs-storage-adapter-for-ghost-blog/88).  I can:
1. Change the default in the storage adapter
2. Change the default in the config file
```js
    if(!!process.env.IPFS_GATEWAY_URL) {
      storageOptions.gatewayURL = process.env.IPFS_GATEWAY_URL;
    } else {
      storageOptions.gatewayURL = "https://ipfs.runfission.com"
    }
```
3. Change the default in app.json
```js
    "IPFS_GATEWAY_URL": {
      "description": "The HTTPS URL of your prefered IPFS gateway.  Defaults to https://ipfs.io/ipfs",
      "value": "https://ipfs.runfission.com",
      "required": false
    }
```
4. Some combination of the above

I'm guessing 3, but the discourse specs were drafted back in June.

**Comment by bmann** [comment]
Yes, fine to edit in just app.json for now. 

The storage adapter won’t work with anything other than our API so someone will have to add support for other APIs or run their own copy of our Web API.

**Comment by MCTaylor17** [comment]
Oops!  This was resolved in commit caa991b
